src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.c

```diff
@@ -217,238 +245,248 @@ void ObjKendoKanban_Destroy(Actor* thisx, PlayState* play) {
-    Vec3f sp48;
+    Vec3f vecCenterOut = this->actor.world.pos;
     s32 pad2;
-    s32 index = 0;
+    s32 indexRootCornerPos = 0;
     s32 i;
-    f32 sp38;
-
-    sp5C = this->unk_26C[0];
-    sp48 = this->actor.world.pos;
-
-    sp48.x -= this->unk_2D8.x;
-    sp48.y -= this->unk_2D8.y;
-    sp48.z -= this->unk_2D8.z;
-
-    sp38 = (this->unk_2F0.x * sp48.z) + (this->unk_2F0.z * -sp48.x);
-
-    if (sp38 < 0.0f) {
-        this->unk_304 += 0x64;
+    f32 verticalScalar;
+
+    // Calculate an affect on the rotation from gravity.
+    vecCenterOut.x -= this->centerPos.x;
+    vecCenterOut.y -= this->centerPos.y;
+    vecCenterOut.z -= this->centerPos.z;
+    verticalScalar = (this->rotationalAxis.x * vecCenterOut.z) + (this->rotationalAxis.z * -vecCenterOut.x);
+    if (verticalScalar < 0.0f) {
+        this->rotationVelocity += 0x64;
     } else {
-        this->unk_304 -= 0x64;
+        this->rotationVelocity -= 0x64;
     }
 
-    for (i = 0; i < ARRAY_COUNT(this->unk_26C); i++) {
-        if (this->unk_26C[i].y < sp5C.y) {
-            sp5C = this->unk_26C[i];
-            index = i;
+    // Find the lowest point
+    for (i = 0; i < ARRAY_COUNT(this->cornerPos); i++) {
+        if (this->cornerPos[i].y < rootCornerPos.y) {
+            rootCornerPos = this->cornerPos[i];
+            indexRootCornerPos = i;
         }
     }
 
-    if (index != this->unk_2FC) {
-        this->unk_300 = 1;
-        this->unk_2FC = index;
-        this->unk_2E4 = this->unk_29C[index];
+    // When the lowest point changes, re-initialize the actor position to that point
+    if (indexRootCornerPos != this->indexLastRootCornerPos) {
+        this->hasNewRootCornerPos = true;
+        this->indexLastRootCornerPos = indexRootCornerPos;
+        this->rootCornerPos = this->cornerPoints[indexRootCornerPos];
 
         Matrix_Push();
         Matrix_SetTranslateRotateYXZ(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
                                      &this->actor.shape.rot);
         Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
-        Matrix_MultVec3f(&this->unk_2E4, &this->actor.world.pos);
-        this->actor.world.pos = sp5C;
+        Matrix_MultVec3f(&this->rootCornerPos, &this->actor.world.pos);
+        this->actor.world.pos = rootCornerPos;
         this->actor.prevPos = this->actor.world.pos;
         Matrix_Pop();
     }
 
     Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4);
 
     if (this->actor.bgCheckFlags & 1) {
+        // When on the ground, apply some friction.
         this->actor.velocity.x *= 0.8f;
         this->actor.velocity.z *= 0.8f;
     }
 
-    if (this->unk_300 == 1) {
-        if (this->unk_308 >= 7) {
-            s16 temp_v0_3 = this->unk_302 & 0x3FFF;
+    if (this->hasNewRootCornerPos == true) {
+        if (this->numBounces >= MAX_BOUNCE_COUNT) {
+            s16 deltaRotationAngle = this->rotationAngle & 0x3FFF; // 90 degrees
```

Comment:
This is more like % 0x4000 than actually 90 degrees, though. I'm not saying try to match it like that, just that that's what it's actually doing.

Reply:
Meant to contextualize the value, not the statement, but I can see how that can be ambiguous. I preferred DEG_TO_BINANG(90) instead of 0x3FFF but saw that was frowned upon in the style documentation, so added the degree comment instead. I'll seek a more clear comment or remove entirely.